src_c/surface.c

```diff
@@ -1863,6 +1865,52 @@ surf_blit(pgSurfaceObject *self, PyObject *args, PyObject *keywds)
         src_rect = &temp;
     }
 
+    /*
+    Yes, this code is really unprofessional-looking,
+    but there is actually the fastest way known to
+    make this work. Using `PyObject_SetAttr()` on a
+    pgRect in C is actually slower than this method.
+
+    This code is also slower than Python code itself.
+
+    TODO: Optimize or cleanup implementation.
+    */
+    if (anchor_text == NULL) {
+    }
+    else if (strcmp(anchor_text, "topleft") == 0) {
```

Comment:
A classic C stunt here might be to abuse `switch`.  Something like:

```c
if (anchor_text != NULL) {
  int valid = 0;
  switch(anchor_text[0]) {
    case 't':
      if (strcmp(anchor_text, "topleft") == 0) {
         ...:
         valid = 1;
      } else if (strcmp(anchor_text, "topright") == 0) {
         ...:
         valid = 1;
      }
      break;
    case 'b':
      ...;
      break;
    case 'm':
      ...;
      break;
    case 'c':
      ...;
      break;
    ....
  }
  if (!valid) {
    return RAISE(PyExc_KeyError, "invalid anchor");
  }
}
```

But then, one might hope that the compiler was smart enough to do this already.